### PR TITLE
DM-34857: add DNS name resolution

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

**Comment by r-owen** [comment]
Unfortunately I don't know how to test the domain name resolution. So I've added it, but not a unit test for it.

File: python/lsst/ts/ess/labjack/labjack_data_client.py [hunk 1]
```diff
@@ -122,7 +123,7 @@ def get_config_schema(cls) -> Dict[str, Any]:
   identifier:
     description: >-
         LabJack indentifier:
-        * An IP address if connection_type=TCP
+        * An IP host name or address if connection_type = TCP or WIFI
```
Comment (wvreeven) [inline python/lsst/ts/ess/labjack/labjack_data_client.py:126]: I think `host name or IP address` would be better. In networking nobody refers to `IP host name` to my best of knowledge.

**Review by wvreeven — APPROVED** [review]
Looks good to me.

**Comment by wvreeven** [comment]
Solving an IP address should return that IP address so that's one thing that could be tested. Solving `localhost` should result into `127.0.0.1` so that's another possibility. However, that _**should**_ have been tested by the maintainers of the socket module so I don't think this needs to be tested in our code.

**Comment by r-owen** [comment]
Yes my concern was how to test the new code "in situ". I can think of things that might work if I hacked up the classes to store the converted IP address, but I can't actually use the resolved address so it's not very satisfactory. I decided to keep it simple and skip trying to test it.

**Pull request merged**